sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -260,6 +261,22 @@ def test_save_figures():
 
     shutil.rmtree(examples_dir)
 
+
+def test_zip_notebooks():
+    """Test generated zipfiles are not corrupt"""
+    gallery_conf = build_test_configuration(examples_dir='examples')
+    examples = [fname
+                for fname in sorted(os.listdir(gallery_conf['examples_dir']))
+                if fname.endswith('.py')]
+    sg.python_zip(examples, gallery_conf['examples_dir'],
+                  gallery_conf['gallery_dir'])
+    zipfilename = os.path.join(gallery_conf['gallery_dir'], "_python.zip")
```

Comment:
for the test they are saved in a temporary folder. The function `build_test_configuration` takes care making this folder and setting other variables.